ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CommandTracker.scala

```diff
@@ -318,6 +318,25 @@ private[commands] class CommandTracker[Context](maxDeduplicationTime: () => Opti
     logic -> promise.future
   }
 
+  private def maxTimeoutFromDeduplicationPeriod(
+      deduplicationPeriod: DeduplicationPeriod,
+      maxDeduplicationDuration: JDuration,
+  ) = {
+    val timeoutDuration = deduplicationPeriod match {
+      case DeduplicationPeriod.Empty =>
+        maxDeduplicationDuration
+      case DeduplicationPeriod.DeduplicationTime(duration) =>
+        JDuration.ofSeconds(duration.seconds, duration.nanos.toLong)
+      case DeduplicationPeriod.DeduplicationOffset(
+            _
+          ) => //no way of extracting the duration from here, will be removed soon
+        maxDeduplicationDuration
+    }
+    Instant
+      .now()
+      .plus(timeoutDuration)
```

Comment:
```suggestion
    Instant.now().plus(timeoutDuration)
```

Reply:
You may want to inject `now` here, too, as it's likely to cause issues if you use slightly different definitions throughout a request.

Reply:
I do aggree, but for now I would not change this because we also use Instant.now() to verify the timeout. I will capture it in another PR

Reply:
@SamirTalwar-DA I'm working on another PR to fix this. But cannot identify a position where we can inject the now from without calling Instant.now(). Throughout the flow I could not find an easy way to inject the now. 

And in retrospect it makes sense:
We want to measure the timeout inside the `CommandTracker` so we're only interested in the time it passed since we've acknowledged the command until it times out, and that has use the same definition (which it does). It has no direct correlation with time used in another parts of the request flow.